cpp/models/abm/location.cpp

```diff
@@ -35,6 +35,7 @@ Location::Location(LocationType type, uint32_t index, uint32_t num_cells)
     , m_testing_scheme()
     , m_cells(std::vector<Cell>(num_cells))
 {
+    set_default_capacity(type);
```

Comment:
a `get_default_capacity` function that returns the capacity is cleaner, so `m_capacity = get_default_capacity()`. This has multiple advantages. It makes the effect of the function more visible. You can then even put it in the constructor initialization list, `m_capacity(get_default_capacity())`. The function also doesn't need to be a member function of Location, it can be a free function.